### PR TITLE
[server] Don't re-set owner cookie

### LINKED ISSUE #5718: [dashboard/server] broken condition ensureWorkspaceAuth on start
The following screenshot shows how often the `auth/workspace-cookies` resource is fetched on workspace starts. 

The `ensureWorkspaceAuth` seems to have a [condition](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/src/start/StartWorkspace.tsx#L197) for fetching this, but it's broken because the cooke isn't accessible from JS (`httpOnly`)

⚠️  Each fetch of this resource causes `findByInstanceId` and `findInstanceById` to be executed. 

Screen Shot 2021-09-16 at 09 58 39

cc. @csweichel

### PR DESCRIPTION
**Pull request by csweichel**
## Description
When starting a workspace we're installing the "owner cookie" which is used for ws-proxy's authentication mechanism. The dashboard will ask many times for this cookie. To ease load on the server, we'll only properly process the request when the cookie isn't set yet.

## Related Issue(s)

Fixes #5718

## How to test
Modify server to produce a log message when this happens. Start a workspace

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **csweichel** after the PR has been reviewed.

Associated issue: *#5718*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by geropl — APPROVED** [review]
LGTM and tested (owner token still there :-) )

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 61d2a7ab9d18cfb5eb901b018f703f1f9a50930d</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5744#pullrequestreview-758331413" title="Approved">geropl</a>*

Associated issue: *#5718*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
